### ISSUE TITLE
`test` task is never run when using `mix do` and space

### ISSUE THREAD

**Issue by vic** [elixir-lang/elixir#5551]
### Bug

The `test` task is never run if followed by space

### Environment

* Elixir & Erlang versions (elixir -v): Elixir 1.4.0-rc.1 (16a14c1)
* Operating system: OSX Sierra

### Current behavior

When using `mix do` to execute two tasks `test` and `clean` and an space character is placed before the comma, tests are never run and the shell just hangs.

```shell
mix new foo
cd foo
env MIX_ENV=test mix do test , clean
Compiling 1 file (.ex)
Generated foo app
# *hangs forever tests are never run*
```

### Expected behavior

Expected `test` task to run.
Expected `mix do test , clean` to be equivalent to `mix do test, clean`

### 

Any other task  works correctly even when followed by space, for example, this works:

```shell
env MIX_ENV=test mix do clean , test
```

**Comment by vic** [comment]
oh, btw I could provide a pull-request if this is confirmed to be a bug by the elixir team.

**Comment by josevalim** [comment]
Thank you. I believe this is a bug. Here is the quickest way to reproduce it: `mix test ""`. There is something causing it to go infinite looping.

**Comment by josevalim** [comment]
@vic btw, we generally appreciate pull requests but this seems to be a rather annoying bug so I will go ahead and fix it asap. It is not we are infinite looping but rather we are loading all files in the operating system. :)

**Comment by josevalim** [comment]
@vic I went ahead and fixed one of the bugs, the one that makes `mix test ""` fail. However, `mix do test , clean` still fails but now saying:

> Test patterns did not match any file:

If you would like to send a pull request for this one, it will be welcome. Keep in mind we tried to generally solve this issue in the past, trying this commit ea00314491799ec32fee34d91a22f82ac103a8b8 and later reverting it bd58e0628f9d9ea3ec15f2b8f05753dc2b784c46 (see PR linked in the issue).

Maybe a simple workaround is to ignore the left side of splitting on `","` if the left is empty.
